thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobRunner.java

```diff
@@ -72,32 +70,45 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
     } else {
       detectionJobContext.setAnomalyFunctionSpec(anomalyFunctionSpec);
 
+      DateTime monitoringWindowStartTime = detectionJobContext.getWindowStartTime();
+      DateTime monitoringWindowEndTime = detectionJobContext.getWindowEndTime();
+      boolean needCleanUpMonitoringWindow = (monitoringWindowEndTime == null);
```

Comment:
Why do cleanup  based on this condition? What do you think about doing this all the time?
If you do decide to keep it condition based, please add a comment here to describe why this is being done